hyperspace/parachain/src/chain.rs

```diff
@@ -203,17 +205,108 @@ where
 
 	async fn query_client_message(&self, update: UpdateClient) -> Result<AnyClientMessage, Error> {
 		use api::runtime_types::{
-			pallet_ibc::pallet::Call as IbcCall, parachain_runtime::Call as RuntimeCall,
+			frame_system::EventRecord,
+			pallet_ibc::pallet::{Call as IbcCall, Event as PalletEvent},
+			parachain_runtime::{Call as RuntimeCall, Event},
 		};
+		use pallet_ibc::events::IbcEvent as RawIbcEvent;
 
 		let host_height = update.height();
-		let light_client_height = update.consensus_height();
 
-		// todo:
-		// first query block events at host_height.
-		// next find the event that matches update
-		// get extrinsic that emitted event.
-		// profit.
+		let now = std::time::Instant::now();
+		let block_hash = loop {
+			let maybe_hash = self
+				.para_client
+				.rpc()
+				.block_hash(Some(host_height.revision_height.into()))
+				.await?;
+			match maybe_hash {
+				Some(hash) => break hash,
+				None => {
+					if now.elapsed() > Duration::from_secs(20) {
+						return Err(Error::from("Timeout while waiting for block".to_owned()))
+					}
+					sleep(Duration::from_millis(100)).await;
+				},
+			}
```

Comment:
hmmmm

Reply:
I had this problem of missing blocks at the beginning of implementation. But maybe I was doing something wrong and now there is no such problem.

Reply:
I don't understand why you were having that issue, the event subscription api is using block header subscription internally, before fetching the events from storage, so by the time this function is called the block should be available, you can try it again without this hack and see if it works.